apps/showcase/src/components/training/code-editor-view/code-editor-view.component.ts

```diff
@@ -238,10 +254,14 @@ export class CodeEditorViewComponent implements OnDestroy {
           // Remove link between launch project and terminals
           await this.webContainerService.loadProject(project.files, project.commands, project.cwd);
         }
-        await this.loadNewProject();
-        this.cwd$.next(project?.cwd || '');
+        this.loadNewProject();
+        this.cwd$.next(project.cwd);
       });
     });
+    this.forceReload.subscribe(async () => {
+      await this.cleanAllModelsFromMonaco();
+      await this.loadAllProjectFilesToMonaco();
+    });
     this.form.controls.code.valueChanges.pipe(
       distinctUntilChanged(),
       skip(1),
```

Comment:
`debounceTime` is set to 300, I think we should increase a bit this number to avoid to much write on the webcontainer (that will trigger app (re)build) and (now with this PR) write on the `localStorage`.
I would propose at least 1sec without typing before we store the changes, what do you think ?

Reply:
I'm ok to auto-save after 1s of inactivity, maybe we can also trigger it on `Ctrl+S`